<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestQAudioInput</class>
 <widget class="QMainWindow" name="TestQAudioInput">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1190</width>
    <height>668</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Times New Roman</family>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>TestQAudioInput</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QGroupBox" name="groupBoxDevice">
       <property name="minimumSize">
        <size>
         <width>360</width>
         <height>400</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>300</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="title">
        <string>支持的音频输入设置</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>音频输入设备</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboDevices"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="actualCodecLabel">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>编码格式</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboCodec"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="actualFreqLabel">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>采样率(Hz)</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="comboSampleRate">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="actualChannelLabel">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>通道数</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="comboChannels"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="actualSampleTypeLabel">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>采样类型</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="comboSampleTypes"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="actualSampleSizeLabel">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>采样大小(Bytes)</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QComboBox" name="comboSampleSizes"/>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="actualEndianLabel">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>字节序</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QComboBox" name="comboByteOrder"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;为了处理方便，本程序固定采样方式：&lt;/p&gt;&lt;p&gt;采样格式如下所示：&lt;/p&gt;&lt;p&gt;采样率：8000Hz；&lt;/p&gt;&lt;p&gt;通道数：1；&lt;/p&gt;&lt;p&gt;编码格式：audio/pcm；&lt;/p&gt;&lt;p&gt;每帧字节数：8bits；&lt;/p&gt;&lt;p&gt;采样点数据类型：无符号整型；&lt;/p&gt;&lt;p&gt;字节序类型：低字节序。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::AutoText</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>实时输入音频波形</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QChartView" name="chartView"/>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="LabBufferSize">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>音频输入设备缓冲区大小为：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="LabBlockSize">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>数据块大小为：</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actDeviceTest"/>
   <addaction name="actStart"/>
   <addaction name="actStop"/>
   <addaction name="actQuit"/>
  </widget>
  <action name="actDeviceTest">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/22.ico</normaloff>:/images/22.ico</iconset>
   </property>
   <property name="text">
    <string>测试音频设置</string>
   </property>
   <property name="toolTip">
    <string>测试音频设置</string>
   </property>
   <property name="font">
    <font>
     <family>宋体</family>
     <pointsize>12</pointsize>
    </font>
   </property>
  </action>
  <action name="actStart">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/626.bmp</normaloff>:/images/626.bmp</iconset>
   </property>
   <property name="text">
    <string>开始输入音频</string>
   </property>
   <property name="toolTip">
    <string>开始</string>
   </property>
  </action>
  <action name="actStop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/624.bmp</normaloff>:/images/624.bmp</iconset>
   </property>
   <property name="text">
    <string>停止音频输入</string>
   </property>
   <property name="toolTip">
    <string>停止</string>
   </property>
  </action>
  <action name="actQuit">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/images/132.bmp</normaloff>:/images/132.bmp</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">qchartview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actQuit</sender>
   <signal>triggered()</signal>
   <receiver>TestQAudioInput</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
